src/analysis/normed_space/weak_dual.lean

```diff
@@ -182,3 +200,256 @@ begin
+      exact continuous.smul continuous_const (continuous_apply _), },
+    exact continuous.smul continuous_const (continuous_apply _), },
+  have sin_closed : is_closed ({0} : set 𝕜) := t1_space.t1 0,
+  have preim_cl : is_closed (φ⁻¹' ({0} : set 𝕜)) := is_closed.preimage φ_cont sin_closed,
+  suffices : range (weak_dual.to_Pi 𝕜 E) ⊆ φ⁻¹' ({0} : set 𝕜),
+  { have key := (is_closed.closure_subset_iff preim_cl).mpr this hf,
+    exact sub_eq_iff_eq_add'.mp (sub_eq_zero.mp key), },
+  intros g hg,
+  cases hg with g₀ hg₀,
+  simp only [algebra.id.smul_eq_mul, mem_singleton_iff, norm_eq_zero, mem_preimage],
+  rw [hφ, ← hg₀],
+  dunfold weak_dual.to_Pi,
+  rw add_comm,
+  simp only [algebra.id.smul_eq_mul, continuous_linear_map.map_add, add_sub_cancel,
+             sub_self, continuous_linear_map.map_smul],
+end
+
+/-- Elements of the closure of the range of the embedding
+`weak_dual.to_Pi : weak_dual 𝕜 E → Π (_ : E), 𝕜` can be viewed as linear maps `E → 𝕜`. -/
+def linear_of_mem_closure_range
+  (f : (Π (_ : E), 𝕜)) (hf : f ∈ closure (range (weak_dual.to_Pi 𝕜 E))) :
+  E →ₗ[𝕜] 𝕜 :=
+{ to_fun := f,
+  map_add' := begin
+    intros z₁ z₂,
+    have key := linear_of_mem_closure_range' f hf (1 : 𝕜) (1 : 𝕜) z₁ z₂,
+    rwa [one_smul, one_smul, one_smul 𝕜 _, one_smul 𝕜 _] at key,
+  end,
+  map_smul' := begin
+    intros c z,
+    have key := linear_of_mem_closure_range' f hf c (0 : 𝕜) z (0 : E),
+    rwa [zero_smul, zero_smul, add_zero, add_zero] at key,
+  end, }
+
+lemma linear_of_mem_closure_range_apply
+  (f : (Π (_ : E), 𝕜)) (hf : f ∈ closure (range (@weak_dual.to_Pi 𝕜 _ E _ _ _))) (z : E) :
+  linear_of_mem_closure_range f hf z = f z := rfl
+
+/-- Elements of the closure of the image under `weak_dual.to_Pi : weak_dual 𝕜 E → Π (_ : E), 𝕜` of
+a subset defined by a non-strict bound on the norm still satisfy the same bound. -/
+lemma norm_eval_le_of_mem_closure_norm_eval_le
+  (z : E) (c : ℝ) (f : (Π (_ : E), 𝕜))
+  (hf : f ∈ closure ((weak_dual.to_Pi 𝕜 E) '' {x' : weak_dual 𝕜 E | ∥ x' z ∥ ≤ c})) :
+  ∥ f z ∥ ≤ c :=
+begin
+  suffices : ∀ (ε : ℝ), 0 < ε → ∥ f (z) ∥ ≤ c + ε,
+  { exact le_of_forall_pos_le_add this, },
+  intros ε ε_pos,
+  have nhd := mem_nhds_Pi_normed_field f z ε_pos,
+  have clos := mem_closure_iff_nhds.mp hf _ nhd,
+  cases clos with g hg,
+  simp only [mem_image, mem_inter_eq, mem_set_of_eq] at hg,
+  rcases hg with ⟨tri, ⟨y', ⟨at_z_le, eq_g⟩⟩⟩,
+  have eq : weak_dual.to_Pi 𝕜 E y' z = y' z := rfl,
+  rw [← eq_g, eq] at tri,
+  have key := norm_add_le_of_le tri.le at_z_le,
+  rwa [sub_add_cancel, add_comm] at key,
+end
+
+/-- Elements of the closure of the image under `weak_dual.to_Pi : weak_dual 𝕜 E → Π (_ : E), 𝕜` of
+a polar `polar s` of a neighborhood `s` of the origin are continuous (linear) functions. -/
+lemma continuous_of_mem_closure_polar
+  {s : set E} (s_nhd : s ∈ 𝓝 (0 : E)) (φ : (Π (_ : E), 𝕜))
+  (hφ : φ ∈ closure ((weak_dual.to_Pi 𝕜 E) '' (weak_dual.polar 𝕜 s))) :
+  @continuous E 𝕜 _ _ φ :=
```

Comment:
```suggestion
  continuous φ :=
```
Again, no need for the `@` version, all the info is already there from the knowledge of phi.